Compilation against kernel 5.17 fails

Due to some changes in kernel 5.17 compilation now fails:
```
rtl8821ce>   CC [M]  /build/source/os_dep/linux/ioctl_cfg80211.o
rtl8821ce> /build/source/os_dep/osdep_service.c: In function 'thread_exit':
rtl8821ce> /build/source/os_dep/osdep_service.c:1302:2: error: implicit declaration of function 'complete_and_exit' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Werror=implicit-function-declaration8;;]
rtl8821ce>  1302 |  complete_and_exit(comp, 0);
rtl8821ce>       |  ^~~~~~~~~~~~~~~~~
rtl8821ce> /build/source/os_dep/osdep_service.c: In function 'rtw_change_ifname':
rtl8821ce> /build/source/os_dep/osdep_service.c:2504:21: warning: passing argument 1 of '_rtw_memcpy' discards 'const' qualifier from pointer target type [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdiscarded-qualifiers-Wdiscarded-qualifiers8;;]
rtl8821ce>  2504 |  _rtw_memcpy(pnetdev->dev_addr, adapter_mac_addr(padapter), ETH_ALEN);
rtl8821ce>       |              ~~~~~~~^~~~~~~~~~
rtl8821ce> /build/source/os_dep/osdep_service.c:930:24: note: expected 'void *' but argument is of type 'const unsigned char *'
rtl8821ce>   930 | void _rtw_memcpy(void *dst, const void *src, u32 sz)
rtl8821ce>       |                  ~~~~~~^~~
rtl8821ce>   CC [M]  /build/source/os_dep/linux/rtw_cfgvendor.o
rtl8821ce>   CC [M]  /build/source/os_dep/linux/wifi_regd.o
rtl8821ce>   CC [M]  /build/source/os_dep/linux/rtw_android.o
rtl8821ce> /build/source/os_dep/linux/ioctl_linux.c: In function 'rtw_mp_efuse_set':
rtl8821ce> /build/source/os_dep/linux/ioctl_linux.c:9789:32: warning: passing argument 1 of '_rtw_memcpy' discards 'const' qualifier from pointer target type [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdiscarded-qualifiers-Wdiscarded-qualifiers8;;]
rtl8821ce>  9789 |   _rtw_memcpy(padapter->pnetdev->dev_addr, get_hal_mac_addr(padapter), ETH_ALEN); /* set mac addr to net_device */
rtl8821ce>       |               ~~~~~~~~~~~~~~~~~^~~~~~~~~~
rtl8821ce> In file included from /build/source/include/drv_types.h:27,
rtl8821ce>                  from /build/source/os_dep/linux/ioctl_linux.c:17:
rtl8821ce> /build/source/include/osdep_service.h:296:31: note: expected 'void *' but argument is of type 'const unsigned char *'
rtl8821ce>   296 | extern void _rtw_memcpy(void *dec, const void *sour, u32 sz);
rtl8821ce>       |                         ~~~~~~^~~
rtl8821ce> /build/source/os_dep/linux/os_intfs.c: In function 'rtw_net_set_mac_address':
rtl8821ce> /build/source/os_dep/linux/os_intfs.c:1244:21: warning: passing argument 1 of '_rtw_memcpy' discards 'const' qualifier from pointer target type [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdiscarded-qualifiers-Wdiscarded-qualifiers8;;]
rtl8821ce>  1244 |  _rtw_memcpy(pnetdev->dev_addr, sa->sa_data, ETH_ALEN); /* set mac addr to net_device */
rtl8821ce>       |              ~~~~~~~^~~~~~~~~~
rtl8821ce> In file included from /build/source/include/drv_types.h:27,
rtl8821ce>                  from /build/source/os_dep/linux/os_intfs.c:17:
rtl8821ce> /build/source/include/osdep_service.h:296:31: note: expected 'void *' but argument is of type 'const unsigned char *'
rtl8821ce>   296 | extern void _rtw_memcpy(void *dec, const void *sour, u32 sz);
rtl8821ce>       |                         ~~~~~~^~~
rtl8821ce> /build/source/os_dep/linux/os_intfs.c: In function 'rtw_os_ndev_register':
rtl8821ce> /build/source/os_dep/linux/os_intfs.c:1699:18: warning: passing argument 1 of '_rtw_memcpy' discards 'const' qualifier from pointer target type [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdiscarded-qualifiers-Wdiscarded-qualifiers8;;]
rtl8821ce>  1699 |  _rtw_memcpy(ndev->dev_addr, adapter_mac_addr(adapter), ETH_ALEN);
rtl8821ce>       |              ~~~~^~~~~~~~~~
rtl8821ce> In file included from /build/source/include/drv_types.h:27,
rtl8821ce>                  from /build/source/os_dep/linux/os_intfs.c:17:
rtl8821ce> /build/source/include/osdep_service.h:296:31: note: expected 'void *' but argument is of type 'const unsigned char *'
rtl8821ce>   296 | extern void _rtw_memcpy(void *dec, const void *sour, u32 sz);
rtl8821ce>       |                         ~~~~~~^~~
rtl8821ce>   CC [M]  /build/source/os_dep/linux/rtw_proc.o
rtl8821ce> cc1: some warnings being treated as errors
rtl8821ce>   CC [M]  /build/source/os_dep/linux/rtw_rhashtable.o
rtl8821ce> make[2]: *** [/nix/store/nc77v0hhn9kp6fdayl6jlahrk0nc4ir0-linux-5.17-dev/lib/modules/5.17.0/source/scripts/Makefile.build:288: /build/source/os_dep/osdep_service.o] Error 1
rtl8821ce> make[2]: *** Waiting for unfinished jobs....
rtl8821ce> /build/source/os_dep/linux/rtw_proc.c: In function 'rtw_drv_proc_open':
rtl8821ce> /build/source/os_dep/linux/rtw_proc.c:215:27: error: implicit declaration of function 'PDE_DATA'; did you mean 'NODE_DATA'? [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Werror=implicit-function-declaration8;;]
rtl8821ce>   215 |  ssize_t index = (ssize_t)PDE_DATA(inode);
rtl8821ce>       |                           ^~~~~~~~
rtl8821ce>       |                           NODE_DATA
rtl8821ce> cc1: some warnings being treated as errors
rtl8821ce> make[2]: *** [/nix/store/nc77v0hhn9kp6fdayl6jlahrk0nc4ir0-linux-5.17-dev/lib/modules/5.17.0/source/scripts/Makefile.build:288: /build/source/os_dep/linux/rtw_proc.o] Error 1
rtl8821ce> make[1]: *** [/nix/store/nc77v0hhn9kp6fdayl6jlahrk0nc4ir0-linux-5.17-dev/lib/modules/5.17.0/source/Makefile:1831: /build/source] Error 2
rtl8821ce> make[1]: Leaving directory '/nix/store/nc77v0hhn9kp6fdayl6jlahrk0nc4ir0-linux-5.17-dev/lib/modules/5.17.0/build'
rtl8821ce> make: *** [Makefile:2244: modules] Error 2
```

I have a similar problem. After updating to Fedora 36 the wireless connection disappeared and the module build fails. Here is my make.log:

`DKMS make.log for rtl8821ce-v5.5.2_34066.20200325 for kernel 5.17.1-300.fc36.x86_64 (x86_64)
Tue Mar 29 06:28:21 PM -03 2022
make ARCH=x86_64 CROSS_COMPILE= -C /lib/modules/5.17.1-300.fc36.x86_64/build M=/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build  modules
make[1]: Entering directory '/usr/src/kernels/5.17.1-300.fc36.x86_64'
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_cmd.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_security.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_debug.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_io.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ioctl_query.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ioctl_set.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ieee80211.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_mlme.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_mlme_ext.o
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_mlme.c: In function ‘rtw_drv_scan_by_self’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_mlme.c:3161:17: warning: this ‘else’ clause does not guard... [-Wmisleading-indentation]
 3161 |                 else
      |                 ^~~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_mlme.c:3164:25: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘else’
 3164 |                         goto exit;
      |                         ^~~~
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_mi.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_wlan_util.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_vht.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_pwrctrl.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rf.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_chplan.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_recv.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_sta_mgt.o
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rf.c: In function ‘dump_txpwr_lmt’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rf.c:765:9: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
  765 |         if (IS_HARDWARE_TYPE_JAGUAR_AND_JAGUAR2(adapter))
      |         ^~
In file included from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:60,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rf.c:17:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/rtw_debug.h:220:9: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
  220 |         do {\
      |         ^~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rf.c:767:17: note: in expansion of macro ‘RTW_PRINT_SEL’
  767 |                 RTW_PRINT_SEL(sel, "txpwr_lmt_5g_20_40_ref:0x%02x\n", rfctl->txpwr_lmt_5g_20_40_ref);
      |                 ^~~~~~~~~~~~~
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ap.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/mesh/rtw_mesh.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/mesh/rtw_mesh_pathtbl.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/mesh/rtw_mesh_hwmp.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_xmit.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_p2p.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rson.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_tdls.o
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_sta_mgt.c: In function ‘rtw_mfree_stainfo’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_sta_mgt.c:379:25: warning: the comparison will always evaluate as ‘true’ for the address of ‘lock’ will never be NULL [-Waddress]
  379 |         if (&psta->lock != NULL)
      |                         ^~
In file included from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:108,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_sta_mgt.c:17:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/sta_info.h:263:17: note: ‘lock’ declared here
  263 |         _lock   lock;
      |                 ^~~~
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_br_ext.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_iol.o
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ap.c: In function ‘rtw_ap_update_chbw_by_ifbmp’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ap.c:4392:42: warning: the comparison will always evaluate as ‘true’ for the address of ‘padapters’ will never be NULL [-Waddress]
 4392 |                 if (!(ifbmp & BIT(i)) || !dvobj->padapters)
      |                                          ^
In file included from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ap.c:17:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:1041:19: note: ‘padapters’ declared here
 1041 |         _adapter *padapters[CONFIG_IFACE_NUMBER];/*IFACE_ID_MAX*/
      |                   ^~~~~~~~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ap.c:4412:42: warning: the comparison will always evaluate as ‘true’ for the address of ‘padapters’ will never be NULL [-Waddress]
 4412 |                 if (!(ifbmp & BIT(i)) || !dvobj->padapters)
      |                                          ^
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:1041:19: note: ‘padapters’ declared here
 1041 |         _adapter *padapters[CONFIG_IFACE_NUMBER];/*IFACE_ID_MAX*/
      |                   ^~~~~~~~~
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_sreset.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_btcoex_wifionly.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_btcoex.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_beamforming.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_odm.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rm.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rm_fsm.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/efuse/rtw_efuse.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/osdep_service.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/os_intfs.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/pci_intf.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/pci_ops_linux.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_linux.o
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/efuse/rtw_efuse.c: In function ‘rtw_efuse_analyze’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/efuse/rtw_efuse.c:781:17: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
  781 |                 if (i % 16 == 0)
      |                 ^~
In file included from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:60,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/efuse/rtw_efuse.c:17:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/rtw_debug.h:231:9: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
  231 |         do {\
      |         ^~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/efuse/rtw_efuse.c:783:25: note: in expansion of macro ‘_RTW_PRINT_SEL’
  783 |                         _RTW_PRINT_SEL(RTW_DBGDUMP, "%02X%s"
      |                         ^~~~~~~~~~~~~~
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/xmit_linux.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/mlme_linux.o
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/osdep_service.c: In function ‘thread_exit’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/osdep_service.c:1302:9: error: implicit declaration of function ‘complete_and_exit’ [-Werror=implicit-function-declaration]
 1302 |         complete_and_exit(comp, 0);
      |         ^~~~~~~~~~~~~~~~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/osdep_service.c: In function ‘rtw_change_ifname’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/osdep_service.c:2504:28: warning: passing argument 1 of ‘_rtw_memcpy’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 2504 |         _rtw_memcpy(pnetdev->dev_addr, adapter_mac_addr(padapter), ETH_ALEN);
      |                     ~~~~~~~^~~~~~~~~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/osdep_service.c:930:24: note: expected ‘void *’ but argument is of type ‘const unsigned char *’
  930 | void _rtw_memcpy(void *dst, const void *src, u32 sz)
      |                  ~~~~~~^~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/os_intfs.c: In function ‘rtw_net_set_mac_address’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/os_intfs.c:1244:28: warning: passing argument 1 of ‘_rtw_memcpy’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1244 |         _rtw_memcpy(pnetdev->dev_addr, sa->sa_data, ETH_ALEN); /* set mac addr to net_device */
      |                     ~~~~~~~^~~~~~~~~~
In file included from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:27,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/os_intfs.c:17:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/osdep_service.h:296:35: note: expected ‘void *’ but argument is of type ‘const unsigned char *’
  296 | extern void     _rtw_memcpy(void *dec, const void *sour, u32 sz);
      |                             ~~~~~~^~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/os_intfs.c: In function ‘rtw_os_ndev_register’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/os_intfs.c:1699:25: warning: passing argument 1 of ‘_rtw_memcpy’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1699 |         _rtw_memcpy(ndev->dev_addr, adapter_mac_addr(adapter), ETH_ALEN);
      |                     ~~~~^~~~~~~~~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/osdep_service.h:296:35: note: expected ‘void *’ but argument is of type ‘const unsigned char *’
  296 | extern void     _rtw_memcpy(void *dec, const void *sour, u32 sz);
      |                             ~~~~~~^~~
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/recv_linux.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_cfg80211.o
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:288: /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/osdep_service.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_linux.c: In function ‘rtw_mp_efuse_set’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_linux.c:9789:46: warning: passing argument 1 of ‘_rtw_memcpy’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 9789 |                 _rtw_memcpy(padapter->pnetdev->dev_addr, get_hal_mac_addr(padapter), ETH_ALEN); /* set mac addr to net_device */
      |                             ~~~~~~~~~~~~~~~~~^~~~~~~~~~
In file included from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:27,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_linux.c:17:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/osdep_service.h:296:35: note: expected ‘void *’ but argument is of type ‘const unsigned char *’
  296 | extern void     _rtw_memcpy(void *dec, const void *sour, u32 sz);
      |                             ~~~~~~^~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_cfg80211.c: In function ‘cfg80211_rtw_scan’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_cfg80211.c:2881:33: warning: the comparison will always evaluate as ‘true’ for the address of ‘ssid’ will never be NULL [-Waddress]
 2881 |                 if (ssids->ssid != NULL
      |                                 ^~
In file included from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/osdep_service_linux.h:93,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/osdep_service.h:50,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:27,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_cfg80211.c:17:
./include/net/cfg80211.h:2211:12: note: ‘ssid’ declared here
 2211 |         u8 ssid[IEEE80211_MAX_SSID_LEN];
      |            ^~~~
make[1]: *** [Makefile:1841: /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build] Error 2
make[1]: Leaving directory '/usr/src/kernels/5.17.1-300.fc36.x86_64'
make: *** [Makefile:2244: modules] Error 2
`

I have a similar problem. After updating to Fedora 36 the wireless connection disappeared and the module build fails. Here is my make.log:

```
DKMS make.log for rtl8821ce-v5.5.2_34066.20200325 for kernel 5.17.1-300.fc36.x86_64 (x86_64)
Tue Mar 29 06:28:21 PM -03 2022
make ARCH=x86_64 CROSS_COMPILE= -C /lib/modules/5.17.1-300.fc36.x86_64/build M=/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build  modules
make[1]: Entering directory '/usr/src/kernels/5.17.1-300.fc36.x86_64'
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_cmd.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_security.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_debug.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_io.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ioctl_query.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ioctl_set.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ieee80211.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_mlme.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_mlme_ext.o
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_mlme.c: In function ‘rtw_drv_scan_by_self’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_mlme.c:3161:17: warning: this ‘else’ clause does not guard... [-Wmisleading-indentation]
 3161 |                 else
      |                 ^~~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_mlme.c:3164:25: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘else’
 3164 |                         goto exit;
      |                         ^~~~
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_mi.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_wlan_util.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_vht.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_pwrctrl.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rf.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_chplan.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_recv.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_sta_mgt.o
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rf.c: In function ‘dump_txpwr_lmt’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rf.c:765:9: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
  765 |         if (IS_HARDWARE_TYPE_JAGUAR_AND_JAGUAR2(adapter))
      |         ^~
In file included from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:60,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rf.c:17:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/rtw_debug.h:220:9: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
  220 |         do {\
      |         ^~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rf.c:767:17: note: in expansion of macro ‘RTW_PRINT_SEL’
  767 |                 RTW_PRINT_SEL(sel, "txpwr_lmt_5g_20_40_ref:0x%02x\n", rfctl->txpwr_lmt_5g_20_40_ref);
      |                 ^~~~~~~~~~~~~
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ap.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/mesh/rtw_mesh.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/mesh/rtw_mesh_pathtbl.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/mesh/rtw_mesh_hwmp.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_xmit.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_p2p.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rson.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_tdls.o
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_sta_mgt.c: In function ‘rtw_mfree_stainfo’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_sta_mgt.c:379:25: warning: the comparison will always evaluate as ‘true’ for the address of ‘lock’ will never be NULL [-Waddress]
  379 |         if (&psta->lock != NULL)
      |                         ^~
In file included from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:108,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_sta_mgt.c:17:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/sta_info.h:263:17: note: ‘lock’ declared here
  263 |         _lock   lock;
      |                 ^~~~
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_br_ext.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_iol.o
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ap.c: In function ‘rtw_ap_update_chbw_by_ifbmp’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ap.c:4392:42: warning: the comparison will always evaluate as ‘true’ for the address of ‘padapters’ will never be NULL [-Waddress]
 4392 |                 if (!(ifbmp & BIT(i)) || !dvobj->padapters)
      |                                          ^
In file included from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ap.c:17:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:1041:19: note: ‘padapters’ declared here
 1041 |         _adapter *padapters[CONFIG_IFACE_NUMBER];/*IFACE_ID_MAX*/
      |                   ^~~~~~~~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_ap.c:4412:42: warning: the comparison will always evaluate as ‘true’ for the address of ‘padapters’ will never be NULL [-Waddress]
 4412 |                 if (!(ifbmp & BIT(i)) || !dvobj->padapters)
      |                                          ^
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:1041:19: note: ‘padapters’ declared here
 1041 |         _adapter *padapters[CONFIG_IFACE_NUMBER];/*IFACE_ID_MAX*/
      |                   ^~~~~~~~~
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_sreset.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_btcoex_wifionly.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_btcoex.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_beamforming.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_odm.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rm.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/rtw_rm_fsm.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/efuse/rtw_efuse.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/osdep_service.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/os_intfs.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/pci_intf.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/pci_ops_linux.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_linux.o
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/efuse/rtw_efuse.c: In function ‘rtw_efuse_analyze’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/efuse/rtw_efuse.c:781:17: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
  781 |                 if (i % 16 == 0)
      |                 ^~
In file included from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:60,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/efuse/rtw_efuse.c:17:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/rtw_debug.h:231:9: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
  231 |         do {\
      |         ^~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/core/efuse/rtw_efuse.c:783:25: note: in expansion of macro ‘_RTW_PRINT_SEL’
  783 |                         _RTW_PRINT_SEL(RTW_DBGDUMP, "%02X%s"
      |                         ^~~~~~~~~~~~~~
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/xmit_linux.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/mlme_linux.o
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/osdep_service.c: In function ‘thread_exit’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/osdep_service.c:1302:9: error: implicit declaration of function ‘complete_and_exit’ [-Werror=implicit-function-declaration]
 1302 |         complete_and_exit(comp, 0);
      |         ^~~~~~~~~~~~~~~~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/osdep_service.c: In function ‘rtw_change_ifname’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/osdep_service.c:2504:28: warning: passing argument 1 of ‘_rtw_memcpy’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 2504 |         _rtw_memcpy(pnetdev->dev_addr, adapter_mac_addr(padapter), ETH_ALEN);
      |                     ~~~~~~~^~~~~~~~~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/osdep_service.c:930:24: note: expected ‘void *’ but argument is of type ‘const unsigned char *’
  930 | void _rtw_memcpy(void *dst, const void *src, u32 sz)
      |                  ~~~~~~^~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/os_intfs.c: In function ‘rtw_net_set_mac_address’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/os_intfs.c:1244:28: warning: passing argument 1 of ‘_rtw_memcpy’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1244 |         _rtw_memcpy(pnetdev->dev_addr, sa->sa_data, ETH_ALEN); /* set mac addr to net_device */
      |                     ~~~~~~~^~~~~~~~~~
In file included from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:27,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/os_intfs.c:17:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/osdep_service.h:296:35: note: expected ‘void *’ but argument is of type ‘const unsigned char *’
  296 | extern void     _rtw_memcpy(void *dec, const void *sour, u32 sz);
      |                             ~~~~~~^~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/os_intfs.c: In function ‘rtw_os_ndev_register’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/os_intfs.c:1699:25: warning: passing argument 1 of ‘_rtw_memcpy’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1699 |         _rtw_memcpy(ndev->dev_addr, adapter_mac_addr(adapter), ETH_ALEN);
      |                     ~~~~^~~~~~~~~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/osdep_service.h:296:35: note: expected ‘void *’ but argument is of type ‘const unsigned char *’
  296 | extern void     _rtw_memcpy(void *dec, const void *sour, u32 sz);
      |                             ~~~~~~^~~
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/recv_linux.o
  CC [M]  /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_cfg80211.o
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:288: /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/osdep_service.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_linux.c: In function ‘rtw_mp_efuse_set’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_linux.c:9789:46: warning: passing argument 1 of ‘_rtw_memcpy’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 9789 |                 _rtw_memcpy(padapter->pnetdev->dev_addr, get_hal_mac_addr(padapter), ETH_ALEN); /* set mac addr to net_device */
      |                             ~~~~~~~~~~~~~~~~~^~~~~~~~~~
In file included from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:27,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_linux.c:17:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/osdep_service.h:296:35: note: expected ‘void *’ but argument is of type ‘const unsigned char *’
  296 | extern void     _rtw_memcpy(void *dec, const void *sour, u32 sz);
      |                             ~~~~~~^~~
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_cfg80211.c: In function ‘cfg80211_rtw_scan’:
/var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_cfg80211.c:2881:33: warning: the comparison will always evaluate as ‘true’ for the address of ‘ssid’ will never be NULL [-Waddress]
 2881 |                 if (ssids->ssid != NULL
      |                                 ^~
In file included from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/osdep_service_linux.h:93,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/osdep_service.h:50,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/include/drv_types.h:27,
                 from /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build/os_dep/linux/ioctl_cfg80211.c:17:
./include/net/cfg80211.h:2211:12: note: ‘ssid’ declared here
 2211 |         u8 ssid[IEEE80211_MAX_SSID_LEN];
      |            ^~~~
make[1]: *** [Makefile:1841: /var/lib/dkms/rtl8821ce/v5.5.2_34066.20200325/build] Error 2
make[1]: Leaving directory '/usr/src/kernels/5.17.1-300.fc36.x86_64'
make: *** [Makefile:2244: modules] Error 2

```

--- a/os_dep/linux/rtw_proc.c
+++ b/os_dep/linux/rtw_proc.c
@@ -39,6 +39,10 @@ inline struct proc_dir_entry *get_rtw_drv_proc(void)
 #define proc_get_parent_data(inode) PDE((inode))->parent->data
 #endif
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 17, 0))
+#define pde_data(inode) PDE_DATA(inode)
+#endif
+
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 24))
 #define get_proc_net proc_net
 #else
@@ -212,7 +216,7 @@ const int drv_proc_hdls_num = sizeof(drv_proc_hdls) / sizeof(struct rtw_proc_hdl
 static int rtw_drv_proc_open(struct inode *inode, struct file *file)
 {
 	/* struct net_device *dev = proc_get_parent_data(inode); */
-	ssize_t index = (ssize_t)PDE_DATA(inode);
+	ssize_t index = (ssize_t)pde_data(inode);
 	const struct rtw_proc_hdl *hdl = drv_proc_hdls + index;
 	void *private = NULL;
 
@@ -234,7 +238,7 @@ static int rtw_drv_proc_open(struct inode *inode, struct file *file)
 
 static ssize_t rtw_drv_proc_write(struct file *file, const char __user *buffer, size_t count, loff_t *pos)
 {
-	ssize_t index = (ssize_t)PDE_DATA(file_inode(file));
+	ssize_t index = (ssize_t)pde_data(file_inode(file));
 	const struct rtw_proc_hdl *hdl = drv_proc_hdls + index;
 	ssize_t (*write)(struct file *, const char __user *, size_t, loff_t *, void *) = hdl->write;
 
@@ -3891,7 +3895,7 @@ const int adapter_proc_hdls_num = sizeof(adapter_proc_hdls) / sizeof(struct rtw_
 
 static int rtw_adapter_proc_open(struct inode *inode, struct file *file)
 {
-	ssize_t index = (ssize_t)PDE_DATA(inode);
+	ssize_t index = (ssize_t)pde_data(inode);
 	const struct rtw_proc_hdl *hdl = adapter_proc_hdls + index;
 	void *private = proc_get_parent_data(inode);
 
@@ -3913,7 +3917,7 @@ static int rtw_adapter_proc_open(struct inode *inode, struct file *file)
 
 static ssize_t rtw_adapter_proc_write(struct file *file, const char __user *buffer, size_t count, loff_t *pos)
 {
-	ssize_t index = (ssize_t)PDE_DATA(file_inode(file));
+	ssize_t index = (ssize_t)pde_data(file_inode(file));
 	const struct rtw_proc_hdl *hdl = adapter_proc_hdls + index;
 	ssize_t (*write)(struct file *, const char __user *, size_t, loff_t *, void *) = hdl->write;
 
@@ -4078,7 +4082,7 @@ const int odm_proc_hdls_num = sizeof(odm_proc_hdls) / sizeof(struct rtw_proc_hdl
 
 static int rtw_odm_proc_open(struct inode *inode, struct file *file)
 {
-	ssize_t index = (ssize_t)PDE_DATA(inode);
+	ssize_t index = (ssize_t)pde_data(inode);
 	const struct rtw_proc_hdl *hdl = odm_proc_hdls + index;
 	void *private = proc_get_parent_data(inode);
 
@@ -4100,7 +4104,7 @@ static int rtw_odm_proc_open(struct inode *inode, struct file *file)
 
 static ssize_t rtw_odm_proc_write(struct file *file, const char __user *buffer, size_t count, loff_t *pos)
 {
-	ssize_t index = (ssize_t)PDE_DATA(file_inode(file));
+	ssize_t index = (ssize_t)pde_data(file_inode(file));
 	const struct rtw_proc_hdl *hdl = odm_proc_hdls + index;
 	ssize_t (*write)(struct file *, const char __user *, size_t, loff_t *, void *) = hdl->write;
 
@@ -4237,7 +4241,7 @@ const int mcc_proc_hdls_num = sizeof(mcc_proc_hdls) / sizeof(struct rtw_proc_hdl
 
 static int rtw_mcc_proc_open(struct inode *inode, struct file *file)
 {
-	ssize_t index = (ssize_t)PDE_DATA(inode);
+	ssize_t index = (ssize_t)pde_data(inode);
 	const struct rtw_proc_hdl *hdl = mcc_proc_hdls + index;
 	void *private = proc_get_parent_data(inode);
 
@@ -4259,7 +4263,7 @@ static int rtw_mcc_proc_open(struct inode *inode, struct file *file)
 
 static ssize_t rtw_mcc_proc_write(struct file *file, const char __user *buffer, size_t count, loff_t *pos)
 {
-	ssize_t index = (ssize_t)PDE_DATA(file_inode(file));
+	ssize_t index = (ssize_t)pde_data(file_inode(file));
 	const struct rtw_proc_hdl *hdl = mcc_proc_hdls + index;
 	ssize_t (*write)(struct file *, const char __user *, size_t, loff_t *, void *) = hdl->write;
 

--- a/os_dep/osdep_service.c
+++ b/os_dep/osdep_service.c
@@ -1299,7 +1299,13 @@ u32 _rtw_down_sema(_sema *sema)
 inline void thread_exit(_completion *comp)
 {
 #ifdef PLATFORM_LINUX
-	complete_and_exit(comp, 0);
+
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 17, 0))
+    kthread_complete_and_exit(comp, 0);
+#else
+    complete_and_exit(comp, 0);
+#endif
+
 #endif
 
 #ifdef PLATFORM_FREEBSD

